### PR TITLE
mrc-4160_test Some fixes to plugin and updates

### PR DESCRIPTION
**Pull request by EmmaLRussell**
This branch adds some fixes to the base branch - you should see that the example index.html now renders and behaves correctly, including updating the readonly form with changes made in the top form. 

Key changes are:
- `plugin.js` needs to register all components with the app, including child components of the DynamicForm. The plugin is called in `index.html` before the form is mounted. I'm not sure if this is the recommended approach - I adapted the implementation [here](https://blog.logrocket.com/building-vue-3-component-library/).  
- dealing with the fact that Vue3 no longer supports making the `v-model` prop of a component configurable in a `model` section. You can still use any prop with `v-model` but you need to annotate with the required property in the component's parent e.g. `v-model:form-meta`, and the component needs to emit an event of the form `update:[propertyName]` e.g. `update:formMeta`. So parents of dynamic forms will need to use `v-model:form-meta` rather than just `v-model`, and that same pattern is used within the package too.
- removed the various uses of `reactive` - these weren't working, I think because when the props are updated, you get a totally new object for whatever prop the component is looking at e.g. `controlGroups`, `formControl` - so initialising a `reactive` on the initial value fails to update following a change. You could use a `computed` instead, but in most case that doesn't make things less verbose as you'd just be replacing `props.whatever` with `whatever.value` - but I have used `computed` in cases where a potentially null value should be replaced with an empty array. 
- reinstated the `watch` on `disabled` in `DynamicForm` so `validate` events can be emitted when required. 

What I haven't looked at:
- using the form via npm - there might be similar issues with registration of components. Probably worth testing outside of HINT
- There are some warnings being logged to console related to the treeselect:
```
vue.global.js:1661 [Vue warn]: Failed to resolve component: transition
If this is a native custom element, make sure to exclude it from component resolution via compilerOptions.isCustomElement. 
  at <VueTreeselectMenu key=1 ref="menu" > 
  at <VueTreeselect multiple=true clearable=false aria-label="Area level"  ... > 
  at <DynamicFormMultiSelect group-label="Area level" formControl= Object onUpdate:formControl=fn  ... > 
  at <BCol md="6" onMousedown=fn<confirm> onClick=fn<confirm> > 
  at <DynamicFormControl key="area_level" group-label="Area level" form-control= Object  ... > 
  at <BRow key=0 class="my-2" > 
  at <DynamicFormControlGroup key=1 control-group= Object readonly=false  ... > 
  at <BCollapse modelValue=true onUpdate:modelValue=fn > 
  at <DynamicFormControlSection key=0 control-section= Object readonly=false  ... > 
  at <BForm ref="d-form" id="d-form" class="dynamic-form col-12"  ... > 
  at <DynamicForm class="col-12" form-meta= Object onUpdate:formMeta=fn<onUpdate:formMeta>  ... > 
  at <App>
```

I'll leave those up to you! :)

### CODE REVIEW

File: src/DynamicFormControl.vue [hunk 1]
```diff
@@ -62,6 +58,7 @@
                     return props.formControl
                 },
                 set(newVal: DynamicControl | undefined) {
+                    console.log("value has changed to" + JSON.stringify(newVal))
```
Comment (LekanAnni) [inline src/DynamicFormControl.vue:61]: rogue, not to worry I will remove it.

File: src/plugin.js [hunk 2]
```diff
@@ -1,4 +1,27 @@
...
 
-Vue.createApp('vue-dynamic-form', DynamicForm);
+const components = {
+    "VueDynamicForm": DynamicForm,
+    DynamicFormControl,
+    DynamicFormControlGroup,
+    DynamicFormControlSection,
+    DynamicFormMultiSelect,
+    DynamicFormNumberInput,
+    DynamicFormSelect
```
Comment (LekanAnni) [inline src/plugin.js:16]: Do we need to export these ?

File: src/plugin.js [hunk 3]
```diff
@@ -1,4 +1,27 @@
...
+
+const plugin = {
+    install (app) {
+        for (const prop in components) {
+            app.component(prop, components[prop]);
+        }
+    }
+}
+
+export default plugin;
```
Comment (LekanAnni) [inline src/plugin.js:27]:
```suggestion
export default {
    install: (app) => {
        app.component('vue-dynamic-form', DynamicForm)
    }
}
```

This works fine.

**Review by LekanAnni — CHANGES_REQUESTED** [review]
Thanks, I have no idea plugin is a feature in vue and that been upgraded too. I will take it from here.

File: src/plugin.js (hunk 2 above)
Comment (EmmaLRussell) [reply src/plugin.js:16]: They're exported in index.ts I think

File: src/plugin.js (hunk 3 above)
Comment (EmmaLRussell) [reply src/plugin.js:27]: Hm, I thought I'd tried it just with the top level component, and it didn't render the lower-level ones into the page. I may have had other issues at that point though..

**Pull request merged**